jth-tests/src/test/java/com/redhat/jenkins/nodesharing/PoolTest.java

```diff
@@ -138,6 +139,58 @@ public void readConfigFromRepo() {
         assertFalse(Pool.ADMIN_MONITOR.getErrors().toString(), Pool.ADMIN_MONITOR.isActivated());
     }
 
+    @Test
+    public void getCredentialsFromJenkins() throws Exception {
+        GitClient cr = j.getConfigRepo();
+        FilePath j2Config = cr.getWorkTree().child("jenkinses").child("jenkins2");
+        StringBuilder newConfig = new StringBuilder(j2Config.readToString());
+        newConfig.append("credential_id=rest-cred-id");
+        j2Config.write(newConfig.toString(), Charset.defaultCharset().name());
+        cr.add("*");
+        cr.commit("Setup");
+        Updater.getInstance().doRun();
+
+        Pool pool = Pool.getInstance();
+        Map<String, String> config = pool.getConfig().getConfig();
+
+        ExecutorJenkins[] jenkinses = pool.getConfig().getJenkinses().toArray(new ExecutorJenkins[0]);
+        assertEquals(2, jenkinses.length);
+        assertNotEquals(jenkinses[0].getCredentialId(), jenkinses[1].getCredentialId());
+
+        UsernamePasswordCredentials creds0 = pool.getExecutorCredential(jenkinses[0]);
+        UsernamePasswordCredentials creds1 = pool.getExecutorCredential(jenkinses[1]);
+
+        assertNotNull(creds0);
+        assertNotNull(creds1);
+        assertNotEquals(creds0, creds1);
+    }
+
+    @Test
+    public void getCredentialsFromJenkinsBroken() throws Exception {
+        GitClient cr = j.getConfigRepo();
+        FilePath j2Config = cr.getWorkTree().child("jenkinses").child("jenkins2");
+        StringBuilder newConfig = new StringBuilder(j2Config.readToString());
+        newConfig.append("credential_id=fake-cred-id");
+        j2Config.write(newConfig.toString(), Charset.defaultCharset().name());
+        cr.add("*");
+        cr.commit("Break it!");
+        Updater.getInstance().doRun();
+
+        Pool pool = Pool.getInstance();
+        boolean found = false;
+        for(ExecutorJenkins jenkins : pool.getConfig().getJenkinses()) {
```

Comment:
Why iterate over the whole ```Set``` and skip all except "Jenkins2"? As in the previous test you can do the same, can't do? Probably you are looking for this ```ExecutorJenkins```:
`ExecutorJenkins ej=Pool.getInstance().getConfig().getJenkinses().toArray(new ExecutorJenkins[0]) [1]`

Reply:
Or ``pool.getConfig().getJenkinsByName(...)`` is legible more...